File: Data Structures & Algorithms/merge-strings-alternately/submission-0.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1 = len(word1)
        l2 = len(word2)

        i = j = 0
        res = ""
        while i < l1 and j < l2:
           res += word1[i]
           i += 1
           res += word2[j]
           j += 1
        if i < l1:
            res += word1[i:]
        if j < l2:
            res += word2[j:]
        return res     

File: Data Structures & Algorithms/find-k-closest-elements/submission-0.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        var res = Arrays.stream(arr)
                .boxed()
                .sorted((a,b) -> {
                    int aDiff = Math.abs(a - x);
                    int bDiff = Math.abs(b - x);
                    if (aDiff < bDiff || (aDiff == bDiff && a < b)) {
                        return -1;
                    } else if (aDiff == bDiff && a.equals(b)) {
                        return 0;
                    } else {
                        return 1;
                    }
                })
                .limit(k)
                .sorted()
                .toList();

        return res;
    }
}